elrond-wasm-node/src/big_int.rs

```diff
@@ -246,12 +246,28 @@ impl NestedDecode for ArwenBigInt {
 }
 
 impl TopDecode for ArwenBigInt {
+    const TYPE_INFO: TypeInfo = TypeInfo::BigInt;
+
 	fn top_decode<I: TopDecodeInput>(input: I) -> Result<Self, DecodeError> {
-        Ok(ArwenBigInt::from_signed_bytes_be(&*input.into_boxed_slice_u8()))
+        // since can_use_handle is provided constantly,
+        // the compiler is smart enough to only ever expand one of the if branches
+        let (can_use_handle, handle) = input.try_get_big_int_handle();
```

Comment:
Is this the same `try_get_big_int_handle()` you mentioned earlier, in the comment "Unless you're developing elrond-wasm, please ignore."?

Reply:
Yes

Reply:
It abstracts away the special hook shortcuts, the ones that give the big int handle directly.
We used to do it with TYPE_INFO and if-s everywhere, but it's more elegant this way.